cardano-api/src/Cardano/Api/IPC/Monad.hs

```diff
@@ -42,15 +50,15 @@ import qualified Ouroboros.Network.Protocol.LocalStateQuery.Type as Net.Query
 -- In order to make pipelining still possible we can explore the use of Selective Functors
 -- which would allow us to straddle both worlds.
 newtype LocalStateQueryExpr block point query r m a = LocalStateQueryExpr
-  { runLocalStateQueryExpr :: ContT (Net.Query.ClientStAcquired block point query m r) m a
+  { runLocalStateQueryExpr :: ReaderT NodeToClientVersion (ContT (Net.Query.ClientStAcquired block point query m r) m) a
```

Comment:
The `ReaderT` allows the monad expression to know the node to client version of the node server.